account_invoice_product_supplier_price_update/wizard/supplier_invoice_update_product_supplier_price.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, fields
+import openerp.addons.decimal_precision as dp
+
+
+class SupplierinvoiceUpdateProductSupplierprice(models.TransientModel):
+    _name = 'supplierinvoice.update.product.supplierprice'
+
+    wizard_line_ids = fields.One2many(
+        'supplierinvoice.update.product.supplierprice.line',
+        'wizard_id',
+        string='Wizard lines')
+
+    @api.multi
+    def update_product_supplierprice(self):
+        invoice = self.env['account.invoice'].browse(
+            self._context['active_id'])
+        invoice.signal_workflow('invoice_open')
+        for line in self.wizard_line_ids:
+            if not line.suppinfo_id:
+                vals = {
+                    'name': line.supplier_id.id,
+                    'min_qty': 0.0,
+                    'delay': 1,
+                }
+                if line.to_variant:
+                    vals.update({
+                        'product_id': line.product_id.id,
+                    })
+                else:
+                    vals.update({
+                        'product_tmpl_id': line.product_id.product_tmpl_id.id,
+                    })
+                line.suppinfo_id = self.env['product.supplierinfo'].create(
```

Comment:
can you avoid the repetition of self.env['product.supplierinfo'] by creating a variable called supinfo_obj ?